src/main/noise/MLS.cpp

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright (C) 2021 Linux Studio Plugins Project <https://lsp-plug.in/>
+ *           (C) 2021 Stefano Tronci <stefano.tronci@protonmail.com>
+ *
+ * This file is part of lsp-dsp-units
+ * Created on: 16 May 2021
+ *
+ * lsp-plugins is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * lsp-plugins is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with lsp-plugins. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <lsp-plug.in/dsp-units/noise/MLS.h>
+
+namespace lsp
+{
+    namespace dspu
+    {
+
+        /** From the table "Exponent of Terms of Primitive Binary Polynomials"
+         *  in Primitive Binary Polynomials by Wayne Stahnke,
+         *  Mathematics of Computation, Volume 27, Number 124, October 1973
+         *  link: https://www.ams.org/journals/mcom/1973-27-124/S0025-5718-1973-0327722-7/S0025-5718-1973-0327722-7.pdf
+         */
+        static const size_t nMaxBits = sizeof(MLS::mls_t) * 8;
+
+        static const MLS::mls_t vTapsMaskTable[] = {
+            // These first 32 elements would use the condition:
+            // #if defined(ARCH_32BIT) || defined(ARCH_64BIT) || defined(ARCH_128BIT)
+            // but all platforms have 32 bit support, so we can simply omit.
+                1u,
```

Comment:
I think it would be easier to read code if we group numbers by 4 or 8 values in the same line.